infra/lambda/datasets-endpoint/test_function.py

```diff
@@ -0,0 +1,330 @@
+"""
+Dataset endpoint Lambda function tests. Working Data Lake AWS environment is
+required (run '$ cdk deploy' before running tests).
+"""
+
+import logging
+import re
+
+import function  # pylint:disable=import-error
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_should_fail_if_request_not_containing_method():
+    """Test if request fails correctly if not containing method attribute."""
+
+    body = {}
+
+    resp = function.lambda_handler({"body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'httpMethod' is a required property."
+
+
+def test_should_fail_if_request_not_containing_body():
+    """Test if request fails correctly if not containing body."""
+
+    method = "POST"
+
+    resp = function.lambda_handler({"httpMethod": method}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'body' is a required property."
+
+
+def test_should_create_dataset(db_prepare):  # pylint:disable=unused-argument
+    """Test Dataset creation using POST method."""
+
+    method = "POST"
+    body = {}
+    body["type"] = "RASTER"
+    body["title"] = "Dataset 123"
+    body["owning_group"] = "A_ABC_XYZ"
+
+    resp = function.lambda_handler({"httpMethod": method, "body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 201
+    assert resp["body"]["type"] == body["type"]
+    assert resp["body"]["title"] == body["title"]
+    assert resp["body"]["owning_group"] == body["owning_group"]
+
+
+def test_should_fail_if_post_request_not_containing_mandatory_attribute():
+    """Test if POST request fails correctly if not containing mandatory attribute."""
```

Comment:
Since the test names are much more descriptive now it looks like these comments are redundant.

Reply:
For now "DB Setup" and "Teardown" are running for each test to make them independent. This is of course slower in case of running all tests.

Reply:
Did you mean to reply to [another comment](https://github.com/linz/geospatial-data-lake/pull/38#discussion_r515662377)?

Reply:
yes, sorry.

Reply:
> Since the test names are much more descriptive now it looks like these comments are redundant.

I think not having docstring require disabling this pylint check probably for whole file :)

Reply:
I'd be happy to disable this check for the whole project, as I consider it an anti-pattern:

- Community guidelines usually say that comments should be about the *why* and not the *what.*
- With descriptive method names, parameter names and parameter types (hopefully to be introduced later) the purpose of the vast majority of methods should be pretty obvious at a glance.
- If it doesn't clarify anything it's just busywork.